Classes/Event/EnrichScriptTagEvent.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Event;
+
+use Brotkrueml\MatomoIntegration\Exceptions\InvalidDataAttributeName;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * This event allows the developer to enrich the build script tag with additional attributes.
+ */
+final class EnrichScriptTagEvent
+{
+    /**
+     * The server request in case the developer requires additional information.
+     */
+    private ServerRequestInterface $serverRequest;
+
+    /**
+     * The id of the element
+     */
+    private string $id = '';
+
+    /**
+     * The type that should be set for the script tag. I.e. text/plain
```

Comment:
e.g. instead of i.e. If we use comments I would propose to explain why text/plain, so something like:

    e.g. text/plain to not execute script on page load, but a consent manager should handle this.